<?xml version="1.0"?>
<robot name="NIFTi">
    <link name="base_link"/>

    <link name="left_track"/>
    <joint name="left_track_j" type="revolute">
        <parent link="base_link"/>
        <child link="left_track"/>
        <origin xyz="0 0.1985 0"/>
        <axis xyz="0 1 0"/>
        <limit lower="-0.7" upper="0.7" effort="0" velocity="0" />
    </joint>

    <link name="right_track"/>
    <joint name="right_track_j" type="revolute">
        <parent link="base_link"/>
        <child link="right_track"/>
        <origin xyz="0 -0.1985 0"/>
        <axis xyz="0 1 0"/>
        <limit lower="-0.7" upper="0.7" effort="0" velocity="0" />
    </joint>

    <link name="front_left_flipper"/>
    <joint name="front_left_flipper_j" type="continuous">
        <parent link="left_track"/>
        <child link="front_left_flipper"/>
        <origin xyz="0.25 0.0735 0.0195" rpy="0 0.1937 0"/>
        <axis xyz="0 1 0"/>
    </joint>

    <link name="front_right_flipper"/>
    <joint name="front_right_flipper_j" type="continuous">
        <parent link="right_track"/>
        <child link="front_right_flipper"/>
        <origin xyz="0.25 -0.0735 0.0195" rpy="0 0.1937 0"/>
        <axis xyz="0 1 0"/>
    </joint>

    <link name="rear_left_flipper"/>
    <joint name="rear_left_flipper_j" type="continuous">
        <parent link="left_track"/>
        <child link="rear_left_flipper"/>
        <origin xyz="-0.25 0.0735 0.0195" rpy="0 2.9479 0"/>
        <axis xyz="0 1 0"/>
    </joint>

    <link name="rear_right_flipper"/>
    <joint name="rear_right_flipper_j" type="continuous">
        <parent link="right_track"/>
        <child link="rear_right_flipper"/>
        <origin xyz="-0.25 -0.0735 0.0195" rpy="0 2.9479 0"/>
        <axis xyz="0 1 0"/>
    </joint>

    <link name="laser"/>
    <joint name="laser_j" type="revolute">
        <parent link="base_link"/>
        <child link="laser"/>
        <origin xyz="0.2502 0.0 0.1427" rpy="3.1415927 0 0"/>
        <axis xyz="1 0 0"/>
        <limit lower="-2.3561945" upper="2.3561945" effort="0" velocity="0" />
    </joint>

    <link name="imu"/>
    <joint name="imu_j" type="fixed">
        <parent link="base_link"/>
        <child link="imu"/>
        <origin xyz="0 0 0.1694" rpy="0 0 0"/>
    </joint>

    <link name="omnicam"/>
    <joint name="omnicam_j" type="fixed">
        <parent link="base_link"/>
        <child link="omnicam"/>
        <origin xyz="0.070 0.068 0.2707" rpy="0 0 -0.942477796"/>
    </joint>

    <link name="camera_0"/>
    <joint name="camera_0_j" type="fixed">
        <parent link="omnicam"/>
        <child link="camera_0"/>
        <origin xyz="0.042089 -0.001704 -0.000358"
                rpy="-1.5717413 -0.0007841 -1.5696985"/>
    </joint>

    <link name="camera_1"/>
    <joint name="camera_1_j" type="fixed">
        <parent link="omnicam"/>
        <child link="camera_1"/>
        <origin xyz="0.011469 -0.04013 -0.000084"
                rpy="-1.5718667 0.000892 -2.8269518"/>
    </joint>

    <link name="camera_2"/>
    <joint name="camera_2_j" type="fixed">
        <parent link="omnicam"/>
        <child link="camera_2"/>
        <origin xyz="-0.034855 -0.02289 0.000523"
                rpy="-1.5725078 0.004121 2.1980575"/>
    </joint>

    <link name="camera_3"/>
    <joint name="camera_3_j" type="fixed">
        <parent link="omnicam"/>
        <child link="camera_3"/>
        <origin xyz="-0.033205 0.025731 0.000218"
                rpy="-1.5713880 0.0056886 0.9420103"/>
    </joint>

    <link name="camera_4"/>
    <joint name="camera_4_j" type="fixed">
        <parent link="omnicam"/>
        <child link="camera_4"/>
        <origin xyz="0.014502 0.038993 -0.0003"
                rpy="-1.5816184 0.0009792 -0.3147923"/>
    </joint>

    <link name="camera_5"/>
    <joint name="camera_5_j" type="fixed">
        <parent link="omnicam"/>
        <child link="camera_5"/>
        <origin xyz="0.000403 -0.000922 0.062129"
                rpy="0.0037705 0.0007895 -1.5718903"/>
    </joint>
</robot>
